src/store/keyboard/context.js

```diff
@@ -0,0 +1,39 @@
+import { createContext, useContext, useState } from 'react';
+
+const KeyboardContext = createContext();
+
+export const KeyboardProvider = ({ children }) => {
+  const [isOptionKey, setIsOptionKey] = useState(false);
+
+  const initialvalue = {
+    isOptionKey,
+    setIsOptionKey,
+  };
+
+  return (
+    <KeyboardContext.Provider value={initialvalue}>
+      <div
+        role="button"
+        tabIndex={0}
+        onKeyDown={(e) => {
+          if (e.altKey) {
+            setIsOptionKey(true);
+          }
+        }}
+        onKeyUp={() => {
+          setIsOptionKey(false);
+        }}
```

Comment:
If we don't check for alt key here I guess it'll setIsOptionKey to false on every key up leading to re-render the components where we have used this context
 
```suggestion
        onKeyUp={() => {
          if (e.altKey)
            setIsOptionKey(false);
        }}
```

Reply:
yes correct, changed it

Reply:
As discussed in vc, e.altKey will always be false in onKeyUp (for any key incl. option key)
If any key is pressed (other than option key) then onKeyUp will set `isOptionKey` to false but it won't lead to re-renders since `isOptionKey` value will already be false